### DISCUSSION TITLE
Read long FL values from DICOM file

### DISCUSSION THREAD

**Discussion by thetestautomator** [fo-dicom/fo-dicom#1775] [Q&A]
Hello together,
I am reading DICOM files using fo-dicom version 5.1.1 but have some issues with FL data.
The files contain elements of VR=FL and values like 3,64713191986084. These data come from a medical devices. When I open the file in an editor like PowerDicom, I can see this value.

If I open this file with fo-dicom, it is shortened to 3,647132
What am I'm missing? 

I read the value like:
```csharp
var dataset = DicomFile.Open("somefile.dcm").Dataset;
float value = dataset.GetSingleValue<float>(DicomTag.SomeTag);

// also tried:
float value2 = dataset.GetDicomItem<DicomFloatingPointSingle>(DicomTag.SomeTag).Get<float>();
```

**Comment by mrbean-bremen** [comment]
`FL` values are [single precision floating point](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) values, meaning that the number of significant digits is limited (about 7, depending on the value). So showing a value of `3,64713191986084` for a single precision (e.g. 4 byte) float may look more precise, but the last digits are in fact meaningless.
So I would say that `fo-dicom` does the right thing here by showing only significant digits.
